jujugui/static/gui/src/app/components/entity-details/content/readme/readme.js

```diff
@@ -93,18 +121,48 @@ YUI.add('entity-content-readme', function() {
     */
     _getReadmeCallback: function(error, data) {
       if (error) {
-        console.error(error); 
+        console.error(error);
         this.setState({readme: 'No readme.'});
       } else {
         var readme = data;
-        this.setState({readme: this.props.renderMarkdown(readme)});
+        const renderMarkdown = this.props.renderMarkdown;
+        let renderer = new renderMarkdown.Renderer();
+        renderer.heading = (text, level) => {
+          const escapedText = text.toLowerCase().replace(/[^\w]+/g, '-');
+          return `<h${level} id="${escapedText}">
+              ${text}
+              <span class="readme-link" data-id="${escapedText}">link</span>
+            </h${level}>`;
+        };
+        this.setState(
+          {readme: renderMarkdown(readme, {renderer: renderer})}, () => {
+            if (this.props.hash) {
+              this._scrollToContent();
+            }
+            // Set up the components to link to the readme headings.
+            document.querySelectorAll('.readme-link').forEach(link => {
```

Comment:
Can this query on this component instead of the document? This will match outside the component which could have some unintended side effects.